app/uk/gov/hmrc/serviceconfigs/connector/DeploymentConfigConnector.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.serviceconfigs.connector
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.StreamConverters
+import play.api.Logging
+import play.api.libs.ws.WSClient
+import uk.gov.hmrc.http.StringContextOps
+import uk.gov.hmrc.serviceconfigs.config.GithubConfig
+import uk.gov.hmrc.serviceconfigs.model.Environment
+
+import java.io.InputStream
+import javax.inject.Inject
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeploymentConfigConnector @Inject()(config: GithubConfig, ws: WSClient)(implicit ec: ExecutionContext,
+                                                                              materializer: Materializer) extends Logging {
+
+  def getAppConfigZip(environment: Environment): Future[InputStream] = {
+    val repoName = s"app-config-${environment.asString}"
+    val url      = url"${config.githubCodeLoadUrl}/hmrc/$repoName/zip/refs/heads/${config.mainBranchName(repoName)}"
+    val key      = config.githubApiOpenConfig.key
+
+    ws.url(url.toString)
+      .withMethod("GET")
+      .withHttpHeaders(("Authorization", s"token ${key}"))
+      .withFollowRedirects(true)
+      .withRequestTimeout(60.seconds)
+      .stream
+      .map(resp => {
+        logger.info(s"Download of ${url} responded with ${resp.status} ${resp.statusText}")
+
+        resp.bodyAsSource.async.runWith(StreamConverters.asInputStream(readTimeout = 60.seconds))
```

Comment:
Is it possible to work with the `akka.Stream` rather than converting to `InputStream`?

Reply:
need the inputstream in order to read the zip. akka-streams/alpakka has some compression support for writing zips but not for reading one. There is some support in more recent versions but i think the implementation unpacks the zip to disk.
There is better support for tar archives, which we could get via the `https://github.com/$user/$repo/tarball/HEAD` link, which might also get around the main branch issue?